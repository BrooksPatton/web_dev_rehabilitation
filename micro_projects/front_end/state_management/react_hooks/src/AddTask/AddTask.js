import React, { useState } from 'react';

function AddTask({ addTask }) {
    const [taskToAdd, setTaskToAdd] = useState('');

    const handleAddTask = event => {
        event.preventDefault();

        addTask(taskToAdd);
        setTaskToAdd('');
    }

    const handleAddTaskChange = event => {
        setTaskToAdd(event.target.value);
    }

    return (
        <section className="AddTask">
            <form onSubmit={handleAddTask}>
                <label>
                    <h1>Add Task</h1>
                    <input type="text" value={taskToAdd} onChange={handleAddTaskChange} />
                </label>
                <button>Add</button>
            </form>
        </section>
    )
}

export default AddTask;