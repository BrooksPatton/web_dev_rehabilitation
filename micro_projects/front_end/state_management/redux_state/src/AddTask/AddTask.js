import React from 'react';

class AddTask extends React.Component {
    constructor(properties) {
        super(properties);
        this.state = {
            taskToAdd: ''
        };
    }

    handleAddTask = event => {
        event.preventDefault();

        this.props.addTask(this.state.taskToAdd);
        this.setState({ taskToAdd: '' });
    }

    handleAddTaskChange = event => {
        this.setState({ taskToAdd: event.target.value });
    }

    render() {
        return (
            <section className="AddTask">
                <form onSubmit={this.handleAddTask}>
                    <label>
                        <h1>Add Task</h1>
                        <input type="text" value={this.state.taskToAdd} onChange={this.handleAddTaskChange} />

                    </label>
                    <button>Add</button>
                </form>
            </section>
        )
    };
}

export default AddTask;