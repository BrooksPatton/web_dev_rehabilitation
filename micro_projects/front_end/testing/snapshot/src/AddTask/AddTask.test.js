import React from 'react';
import renderer from 'react-test-renderer';

import AddTask from './AddTask';

it('should render correctly', () => {
    const addTaskMock = () => { };

    const tree = renderer
        .create(<AddTask addTask={addTaskMock} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});