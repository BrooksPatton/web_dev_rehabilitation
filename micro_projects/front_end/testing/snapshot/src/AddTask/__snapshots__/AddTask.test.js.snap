// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<section
  className="AddTask"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      <h1>
        Add Task
      </h1>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <button>
      Add
    </button>
  </form>
</section>
`;
